### PR TITLE
Don't commit for read-only query

### PR DESCRIPTION
**Pull request by uranusjr**

### CODE REVIEW

File: task_sdk/src/airflow/sdk/definitions/asset/__init__.py [hunk 1]
```diff
@@ -429,10 +429,13 @@ class AssetAlias(BaseAsset):
...
 
     def _resolve_assets(self) -> list[Asset]:
+        from airflow import settings
         from airflow.models.asset import expand_alias_to_assets
-        from airflow.utils.session import create_session
 
-        with create_session() as session:
+        # We don't use create_session or provide_session here because this is
+        # called in the scheduler when commit is prohibited (for HA reasons).
+        # Those functions assume you want to commit the result on exit.
```
Comment (ephraimbuddy) [inline task_sdk/src/airflow/sdk/definitions/asset/__init__.py]: Isn't it possible to pass the scheduler session to this?

**Review by Lee-W — APPROVED** [review]

File: task_sdk/src/airflow/sdk/definitions/asset/__init__.py (hunk 1 above)
Comment (uranusjr) [reply task_sdk/src/airflow/sdk/definitions/asset/__init__.py]: It is, but this function is also (indirectly) used by places where there’s not an active session. I feel it’s easier to open a new one for this (it doesn’t really cost much).

**Comment by uranusjr** [comment]
Why are those things failing…

**Comment by uranusjr** [comment]
I don’t know why it failed, but I found a way that works.

File: task_sdk/src/airflow/sdk/definitions/asset/__init__.py (hunk 1 above)
Comment (uranusjr) [reply task_sdk/src/airflow/sdk/definitions/asset/__init__.py]: I changed this to accept an optional session when it makes sense instead.

**Review by ephraimbuddy — APPROVED** [review]

**Review by shahar1 — APPROVED** [review]

**Pull request merged**